hail/python/test/hail/table/test_table.py

```diff
@@ -206,6 +206,20 @@ def test_to_matrix_table(self):
 
         assert re_mt.choose_cols(mapping).drop('col_idx')._same(mt.drop('col_idx'))
 
+    def test_to_matrix_table_row_major(self):
+        t = hl.utils.range_table(10)
+        t = t.annotate(**{'foo': 1, 'bar': 2, 'baz': 3})
+        mt = t.to_matrix_table_row_major(['bar', 'baz'], 'entry', 'col')
+        t2 = mt.localize_entries('entries', 'cols')
+        t2 = t2.transmute(**{col.col: t2.entries[i].entry for i, col in enumerate(hl.eval(t2.cols))})
+        t2 = t2.drop(t2.cols)
+
+        self.assertTrue(t._same(t2))
```

Comment:
Hmm, I'm worried that this test is six lines. This suggests to me we have a complex interface.

I wonder if we should have a method on MT's that is closer to an inverse.

```python
t = hl.utils.range_table(10)
t = t.annotate(foo=t.idx, bar=t.idx * 2, baz=t.idx * 3)
mt = t.to_matrix_table_row_major(['bar', 'baz'], 'entry', 'col')
expected = t
actual = mt.to_table_of_rows()
expected._sam(actual)
```

I think `to_table_of_rows` is just what you've written here. Of course, this relies on the column key being one field of type string.

Reply:
I think that'd be good to have, since it might help users understand a bit better the row vs entry representations if they can easily switch between them. At that point though I feel like it would be good to have a single PR that explicitly implements all these conversions with a consistent naming scheme.